### PR TITLE
Generic handling of sklearn distance metric

### LINKED ISSUE #1296: Let `KElbowVisualizer` use all the distance metrics supported by sklearn
**Describe the solution you'd like**
Is there a reason `KElbowVisualizer` can use only the distance metrics in `DISTANCE_METRICS`?

The `distortion_score` relies on `sklearn.metrics.pairwise.pairwise_distances()`, that supports distance metrics not included in  `DISTANCE_METRICS`.
For the silhouette score  `KElbowVisualizer` uses `sklearn.metrics.cluster._unsupervised.silhouette_score()`, that also supports metrics not included in  `DISTANCE_METRICS`.

I don't see a reason why there is this limitation.

### PR DESCRIPTION
**Pull request by bbengfort**
This PR fixes #1296  which reported that Yellowbrick was being too restrictive of what distance metrics were allowed to be used with the KElbow visualizer and did not generalize to all scikit-learn distance metrics. 

I have made the following changes:

1. Removed the `DISTANCE_METRICS` string constants
2. Ensured that callables were eligible to be distance metrics 
3. Used `sklearn.metrics.DistanceMetric.get_metric` to validate if the metric was good or not.

There was already a test in place for testing that a `YellowbrickValueError` is raised if a bad metric is passed into the visualizer, and this test is still passing. 

### Sample Code and Plot

Previously we were unable to run the following code, it should now be possible:

```python
viz = KElbowVisualizer(distance_metric="chebyshev") 
viz.fit(X_train, y_train)
viz.show() 
```

### CHECKLIST

- [x] _Is the commit message formatted correctly?_
- [ ] _Have you noted the new functionality/bugfix in the release notes of the next release?_

- [ ] _Included a sample plot to visually illustrate your changes?_
- [x] _Do all of your functions and methods have docstrings?_
- [x] _Have you added/updated unit tests where appropriate?_
- [ ] _Have you updated the baseline images if necessary?_
- [x] _Have you run the unit tests using `pytest`?_
- [x] _Is your code style correct (are you using PEP8, pyflakes)?_
- [ ] _Have you documented your new feature/functionality in the docs?_

- [ ] _Have you built the docs using `make html` (must be run from `docs/`)?_

### CODE REVIEW

**Review by lwgray — APPROVED** [review]
Looks good to me!  Thank you for contributing.  You always write such efficient code.
